### PR TITLE
Add node 20 support to task-lib for testing and pick highest node version for tests

### PR DESCRIPTION
**Pull request by DmitriiBobreshev**
**Description**:
- Added node 20 support
- Changed the logic to pick the highest handler instead of the first one
- Added tests for node20
- Changed tests for node6, node10, node16 to use highest handler

**Related WI**:  2180233

**Testing**:
Tested locally with different tasks with node 16 and node20 handlers

### CODE REVIEW

File: node/test/fakeTasks/node20task/task.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "id": "id",
+    "name": "Node6Task",
```
Comment (KonstantinTyukalov) [inline node/test/fakeTasks/node20task/task.json:3]: `"Node20Task"`?

**Review by KonstantinTyukalov — APPROVED** [review]

File: node/test/fakeTasks/node20task/task.json (hunk 1 above)
Comment (DmitriiBobreshev) [reply node/test/fakeTasks/node20task/task.json]: Fixed, thanks!

**Review by kirill-ivlev — APPROVED** [review]

File: node/mock-test.ts [hunk 2]
```diff
@@ -167,21 +167,17 @@ export class MockTestRunner {
...
-                downloadVersion = 'v10.21.0';
-                break;
-            case 16:
-                downloadVersion = 'v16.13.0';
-                break;
-            default:
-                throw new Error('Invalid node version, must be 6, 10, or 16 (received ' + version + ')');
-        }
+        const versions = {
+            20: 'v20.11.1',
```
Comment (DergachevE) [inline node/mock-test.ts:171]: Is this correct version of Node 20? Shouldn't we use the latest (20.13.1)? @kirill-ivlev

Comment (DmitriiBobreshev) [reply node/mock-test.ts]: We should, when I checked last time the agent used 20.11.1. Fixed, thanks!

File: node/mock-test.ts [hunk 3]
```diff
@@ -203,31 +199,26 @@ export class MockTestRunner {
...
+                const currExecutor = key.toLocaleLowerCase();
+                if (!currExecutor.startsWith('node')) continue;
+                const version = currExecutor.replace('node', '');
+                const intVersion = parseInt(version) || 6; // node handler is node v6 by default
+                nodeVersion = Math.max(intVersion, nodeVersion);
             }
         }
 
-        if (!nodeVersionFound) {
+        if (nodeVersion == 0) {
```
Comment (DergachevE) [inline node/mock-test.ts:216]: nitpick: `===`?

Comment (DmitriiBobreshev) [reply node/mock-test.ts]: fixed, thanks!

**Review by DergachevE — APPROVED** [review]

**Pull request merged**